src/Symfony/Component/Routing/Matcher/UrlMatcher.php

```diff
@@ -259,4 +262,28 @@ protected function createRequest($pathinfo)
             'SCRIPT_NAME' => $this->context->getBaseUrl(),
         ));
     }
+
+    /**
+     * Match all variables enclosed in "{}" and substitute value using related
+     * attribute key.
+     *
+     * @param array $defaults
+     * @param array $attributes
+     */
+    private function resolveDefaults(&$defaults, $attributes)
+    {
+        $enclosingPattern = "/{(\w+)}/";
+
+        foreach ($defaults as $key => $value) {
+            $valueProcessed = preg_replace_callback($enclosingPattern, function ($match) use ($attributes) {
+                if (!isset($attributes[$match[1]])) {
```

Comment:
@voronkovich this configuration has no sense, the default value for name will overwrite the routing value.

```yaml
hello:
    path: /hello/{name}
    defaults:
        hello: 'Hello, {name}!'
        name: 'John'
```

Because the routing should receive always a value.